pkg/apis/options/upstreams.go

```diff
@@ -8,7 +8,15 @@ const (
 )
 
 // Upstreams is a collection of definitions for upstream servers.
-type Upstreams []Upstream
+type Upstreams struct {
```

Comment:
This feels a bit weird to be called `Upstreams` now right? I keep reading `upstreams.configs` and I think it would be better if we had it as `upstream.configs`. 

I wonder if we should rename this to `Upstream` and the current `Upstream` struct to be something like `UpstreamConfig`, this could be a follow up but should be done before we release I think 🤔